pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -868,6 +904,34 @@ func TestDaemonSetPodCreateExpectationsError(t *testing.T) {
 	}
 }
 
+func TestGetPodsForDaemonSetUsesIndexer(t *testing.T) {
```

Comment:
IMO this should be a Benchmark test: https://pkg.go.dev/testing#hdr-Benchmarks

Reply:
Synced offline.

This test falls in the realm of catching correctness and performance regressions.
Benchmarking is useful when you want to run multiple loops and  produce the reports for Indexing VS pure List calls. That is not the purpose of this test.
Purpose of the test is to catch regression when controllers move from Index to LIST.